bot/cogs/wolfram.py

```diff
@@ -0,0 +1,65 @@
+import aiohttp
+import discord
+import os
+from io import BytesIO
+from urllib import parse
+from discord.ext import commands
+from bot.settings import APPID
+
+
+class Wolfram(commands.Cog):
+    """command for wolfram search"""
+    def __init__(self, bot):
+        self.bot = bot
+        self.query = "http://api.wolframalpha.com/v2/{request}?{data}"
+        
+
+    # wolfram command, takes in a search and gives the result
+    # PS, I didn't totally steal it from seasonal bot
+    @commands.command(brief="Takes in a wolfram search and displays the result", usage=".wolfram [query]")
+    async def wolfram(self, ctx, *, query):
+
+        url_str = parse.urlencode({
+            "i": query,
+            "appid": APPID,
+        })
+
+        query_final = self.query.format(request="simple", data=url_str)
+
+        async with ctx.channel.typing():
+            async with aiohttp.ClientSession() as cs:
+                async with cs.get(query_final) as response:
+                    status = response.status
+                    image_bytes = await response.read()
+
+                    image_file = discord.File(BytesIO(image_bytes), filename="image.png")
+                    image_url = "attachment://image.png"
+
+                    if status == 501:
+                        message = "Failed to get response"
+                        footer = ""
+                        color = discord.Color.red()
+                        
+                    elif status == 400:
+                        message = "No input found"
+                        footer = ""
+                                                
+                    elif status == 403:
+                        message = "Wolfram API key is invalid or missing."
+                        footer = ""
+                        
+                    else:
+                        message = ""
+                        footer = "View original for a bigger picture."
+                        color = discord.Colour.orange()
```

Comment:
Instead of using if/elif statements, it would be more efficient to just have the request raise an error if something goes wrong.

```suggestion
                    assert response.status == 200, response.raise_for_status()
                    image_bytes = await response.read()

                    image_file = discord.File(BytesIO(image_bytes), filename="image.png")
                    image_url = "attachment://image.png"

                    message = ""
                    footer = "View original for a bigger picture."
                    color = discord.Colour.orange()
```

Reply:
shouldn't there be an embed for an error? i mean if an error occurs it just shows typing and nothing shows up, which can be confusing for the user

Reply:
@saiTama-max There's an event called `on_command_error` that's currently sitting in `bot.py` with only a `log.info` to show that anything's wrong. I will be updating this soon.

Reply:
ohok so i will commit my changes rn